misk/src/main/kotlin/misk/CoordinatedService.kt

```diff
@@ -57,11 +53,11 @@ internal class CoordinatedService(val service: Service) : AbstractService() {
 
   fun startIfReady() {
     synchronized(this) {
-      if (state() != Service.State.STARTING) return
+      if (state() != State.STARTING || service.state() != State.NEW) return
```

Comment:
can you update the tests to catch these state transition errors?

Reply:
I was thinking about how to do that. I might be able to do it without adding threads to the tests. Currently the test for this is single-threaded so it’s unlikely to be flaky.